webhook_server/tests/test_prepare_retest_welcome_comment.py

```
@@ -95,3 +97,224 @@ def test_prepare_retest_welcome_comment(
         assert pull_request_handler._prepare_retest_welcome_comment == expected
+
+
+class TestWelcomeMessageNewlineStructure:
+    """Regression tests for welcome message markdown formatting.
+
+    These tests ensure that conditional sections in the welcome message
+    maintain proper newline structure to prevent headers from being
+    glued to previous content.
+    """
+
+    def _create_handler(self, process_github_webhook, owners_file_handler, **config):
```

Ruff (0.14.10)

Missing return type annotation for private function `_create_handler`

(ANN202)